### PR TITLE
return UIButton as a result in addSideMenuButton 

### PR DESCRIPTION
**Pull request by nodify-at**
so we can have more flexibility to change the toggle menu button.

### CODE REVIEW

**Comment by teodorpatras** [comment]
I think the closure approach s better. I don't really like this: `let _ = controller.addSideMenuButton()` .. can you add a nullable closure parameter to the function which defaults to nil?

**Comment by nodify-at** [comment]
ok, it would be better. I'll change it today.

**Comment by nodify-at** [comment]
@teodorpatras I've pushed new changes to provide an optional closure instead returning the button.

File: Source/UIKitExtensions.swift [hunk 1]
```diff
@@ -32,7 +32,7 @@ extension UIView {
 }
 
 public extension UINavigationController {
-    public func addSideMenuButton() {
+    public func addSideMenuButton(_ sideMenuButtonAdded: ((UIButton) -> Void)? = {button in}  ) {
```
Comment (teodorpatras) [inline Source/UIKitExtensions.swift]: please change to:  `addSideMenuButton(completion: ((UIButton) -> ())? = nil)`

File: Source/UIKitExtensions.swift [hunk 2]
```diff
@@ -57,6 +57,10 @@ public extension UINavigationController {
         }else{
             self.topViewController?.navigationItem.rightBarButtonItems = [spacer, item]
         }
+        
+        if sideMenuButtonAdded != nil {
```
Comment (teodorpatras) [inline Source/UIKitExtensions.swift]: :) that's not how you do it (have a look [here](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/OptionalChaining.html))

change to `completion?(button)`

**Comment by nodify-at** [comment]
:) thanks for your feedback. I've improved the code as you requested

File: Source/UIKitExtensions.swift (hunk 2 above)
Comment (nodify-at) [inline Source/UIKitExtensions.swift]: changed :)

**Pull request merged**